opal/mca/base/mca_base_component_find.c

```diff
@@ -189,8 +189,9 @@ int mca_base_components_filter (mca_base_framework_t *framework, uint32_t filter
         mca_base_open_only_dummy_component_t *dummy =
             (mca_base_open_only_dummy_component_t *) cli->cli_component;
 
-        can_use = use_component (include_mode, (const char **) requested_component_names,
-                                 cli->cli_component->mca_component_name);
+        can_use = (cli->cli_component->mca_component_flags & MCA_BASE_COMPONENT_FLAG_ALLWAYS_CONSIDER) ||
```

Comment:
not sure if we want to disable the flag if a component is dynamic. the static requirement allows us to check the flag without dlopening the component.